ElrondNetwork/elrond-go
# Remove libp2p reference in go mod

## Reasoning behind the pull request
- remove the libp2p-p2p-core reference in go.mod
  
## Proposed changes
- reuse components from elrond-go-p2p

## Testing procedure
- standard system test
- generate a new p2p key using the key generator, use that key on a new seeder/node instance and check that the seednode/node is started with the provided key (log messages in the node and compare with the contents of the p2pKey.pem file)

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
